### ISSUE TITLE
Ability to use Explanatory Text as label for a field

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4778]
At the UI level, in Form Builder:

- In the Control Settings dialog, in the Label tab, form authors would have the option of either entering a label, as it is the case now, or to select one (or multiple, if #4758 is implemented) Explanatory Text "fields" as providing the label for the current field.
- We could make that selection easier by first suggesting Explanatory Text fields that are on the same line or the same column as the current field, if any.

In the form definition:

- When an Explanatory Text is selected as a label, maybe change the `fr:explanation` into an `fr:label`, as shown below. Having an `fr:label` would be needed by the XForms, but might make things more complex for Form Builder. So, instead, we might want to keep the `fr:explanation` in the form definition, and transform it into an `fr:label` at runtime.
- On the control, add an `aria-labelledby="…"` pointing to the label (we could do this "preemptively" even if for now we support labels referenced by multiple controls or controls referencing multiple labels, so Form Builder doesn't ultimately have support both a `for` on the `fr:explanation` or `xf:label` *and* `aria-labelledby` controls).

```xml
<fr:explanation id="control-2-control" bind="control-2-bind">
    <fr:text ref="$form-resources/control-2/text" mediatype="text/html"/>
</fr:explanation>

<fr:label id="control-2-control" ref="$form-resources/control-2/text" mediatype="text/html"/>
```

[+1 from customer](https://3.basecamp.com/3600924/buckets/1966555/messages/3259104091)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/18345583/messages/5103766740)

**Comment by ebruchez** [comment]
#4758 is now implemented, and with #6645 we also fixed some bugs. So we can implement this more easily.

Correction to the original comment above: we wouldn't use `<fr:label>`, but `<xf:label>`.

At the XForms level, it makes more sense to have:

```xml
<xf:label for="my-control-id">
```

rather than the reverse `aria-labelledby`. The XForms processor will perform that reverse linking with `aria-labelledby`.

If we want to keep `<fr:explanation>` in the form definition, then we could add the `for` to that:

```xml
<fr:explanation id="control-2-control" bind="control-2-bind" for="my-control-id">
    <fr:text ref="$form-resources/control-2/text" mediatype="text/html"/>
</fr:explanation>
```

Alternatively, we could make Form Builder handle the external `<xf:label>` directly. The benefit of this is that if you want an external hint, you use `<xf:hint>`.

In addition, the nested `<xf:label>` within the control must be removed/added as the user switches between local/external label. I think that we want for now to have only one label, either local or external, even though at the lower level it's possible to have more than one label.

**Comment by avernet** [comment]
> In addition, the nested `<xf:label>` within the control must be removed/added as the user switches between local/external label. I think that we want for now to have only one label, either local or external, even though at the lower level it's possible to have more than one label.

I agree: I can't think of a use case that would benefit from supporting both an internal and an external label. However, to clarify, we do want to support multiple external labels, as this was envisioned as a way for Form Builder to support use cases where controls are labeled by both text at the beginning of the row and at the top of the column (e.g., rating, reporting different numbers for different time periods, etc.).

**Comment by ebruchez** [comment]
[+1 from customer](https://mail.google.com/mail/u/0/#inbox/FMfcgzQfCMqdSDljvCjPHGJLxRNFxvzq)
